src/shared/exec-util.c

```diff
@@ -365,8 +366,47 @@ static int gather_environment_consume(int fd, void *arg) {
         return r;
 }
 
+ExecCommandFlags exec_command_flags_from_strv(char **ex_opts) {
+        ExecCommandFlags flags = 0, ex_flag = 0;
+        char **opt = NULL;
+
+        STRV_FOREACH(opt, ex_opts) {
+                if (!isempty(*opt)) {
+                        ex_flag = exec_command_flags_from_string(*opt);
+                        if (ex_flag != (ExecCommandFlags) -1)
+                                flags |= ex_flag;
+                }
+        }
+
+        return flags;
+}
+
+char **exec_command_flags_to_strv(ExecCommandFlags flags) {
+        ExecCommandFlags it = flags;
+        char **ex_opts = NULL;
+        int i;
+
+        for (i = 0; it != 0; it &= ~(1 << i), i++) {
+                if (FLAGS_SET(flags, (1 << i)))
+                        if (strv_extend(&ex_opts, exec_command_flags_to_string((1 << i))) < 0)
+                                log_oom();
```

Comment:
this should not log itself but simply return NULL and let the caller  handle the OOM case here.

Generally, in systemd we try to make sure that functions either log about all their errors or about none, and "API-like" functions (as opposed to "main()-like" functions) should generally be of the latter kind, and thus allow the caller to log. This function is certainly of the API-like kind, and hence really should just propagate the OOM case and that's it.